frontend/src/routes/(authenticated)/project/[project_code]/+page.svelte

```diff
@@ -218,12 +218,33 @@
     await hgCommand(async () => fetch(`/api/project/hgRecover/${project.code}`));
   }
 
+  async function streamHgCommandResponse(body: ReadableStream<Uint8Array> | null): Promise<void> {
+    if (body == null) return;
+    const decoder = new TextDecoder();
+    // Would be nice to just do this:
+    // for await (const chunk of body) {
+    //   hgCommandResponse += decoder.decode(chunk, {stream: true});
+    // }
+    // But that only works on Firefox. So until Chrome implements that, we have to do this:
+    const reader = body.getReader();
+    await reader.read().then(function handleChunk ({ done, value }): Promise<null> | null {
+      // The {stream: true} is important here; without it, in theory the output could be
+      // broken in the middle of a UTF-8 byte sequence and get garbled. But with stream: true,
+      // TextDecoder() will know to expect more output, so if the stream returns a partial
+      // UTF-8 sequence, TextDecoder() will return everything except that sequence and wait
+      // for more bytes before decoding that sequence.
+      hgCommandResponse += decoder.decode(value, {stream: true});
+      return done ? null : reader.read().then(handleChunk);
+    });
```

Comment:
could we do something like this:
let done = false;
let value;
while (!done) {
    {done, value} = await reader.read();
    hgCommandResponse += decoder.decode(value, {stream: true});
}

rather than this complicated self referencing callback?

Reply:
The self-referencing callback is the example code from MDN, copied almost verbatim except for changing the name of the function involved and simplifying it. It's perfectly understandable to me: it's basically a tail-recursive call, just like you'd write in Lisp or F#. But I can change it to the more complicated while loop if you want.

Reply:
Done in commit 52342f60, though I still think that version looks worse than the simple tail-recursive call. (I couldn't do `{done, value} = await reader.read()`, though I just figured out what syntax I was missing. One sec, let me fix that.)

Reply:
Okay, now done in commit 0db37559. The parentheses needed to be around the entire `{done, value}= await promise` statement. I had been putting them just around the `{ done, value }` destructuring bit, which wasn't the correct syntax. Found the correct syntax at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment.